code/game/machinery/transformer.dm

```diff
@@ -27,32 +28,39 @@
 
 /obj/machinery/transformer/examine(mob/user)
 	. = ..()
-	if(cooldown && (issilicon(user) || isobserver(user)))
-		. += "It will be ready in [DisplayTimeText(cooldown_timer - world.time)]."
+	if(!COOLDOWN_FINISHED(src, transform_cooldown))
+		var/timeleft = DisplayTimeText(round(COOLDOWN_TIMELEFT(src, transform_cooldown)))
+		. += "It will be ready in <b>[timeleft]</b>."
 
 /obj/machinery/transformer/Destroy()
 	QDEL_NULL(countdown)
 	. = ..()
 
 /obj/machinery/transformer/update_icon_state()
-	if(machine_stat & (BROKEN|NOPOWER) || cooldown == 1)
+	if(machine_stat & (BROKEN|NOPOWER) || COOLDOWN_FINISHED(src, transform_cooldown))
 		icon_state = "separator-AO0"
 	else
 		icon_state = initial(icon_state)
 
 /obj/machinery/transformer/Bumped(atom/movable/AM)
-	if(cooldown == 1)
+	if(!COOLDOWN_FINISHED(src, transform_cooldown) && COOLDOWN_FINISHED(src, messagespam_cooldown))
+		var/timeleft = DisplayTimeText(round(COOLDOWN_TIMELEFT(src, transform_cooldown)))
+		say("ERROR: Equipment recalibration in progess. Please wait [timeleft].")
+		COOLDOWN_START(src, messagespam_cooldown, MSGSPAM_CD_DURATION)
 		return
```

Comment:
maybe we could move this check to AFTER the ishuman() check and such, so that random pieces of trash won't trigger this message every 20 seconds, and humans in invalid states would trigger more appropriate/relevant/important messages than this one?

Reply:
Trash and stuff will normally just pass through the machine, and there's rarely any reason something other than a human would be dragged on the conveyor belt anyways